fendermint/vm/interpreter/src/fvm/state/query.rs

```diff
@@ -109,7 +111,9 @@ where
         .context("error creating execution state")?;
 
         let res = f(&mut exec_state);
-        *cache = Some(exec_state);
+        if use_cache {
```

Comment:
This should also be used when deciding whether to [use the cache](https://github.com/consensus-shipyard/fendermint/blob/b1493d05b46891c72f43653d747b2ace790c0958/fendermint/vm/interpreter/src/fvm/state/query.rs#L101) or not.